### ISSUE TITLE
Twig fails to install using Composer and PHP 8 when following official docs example

### ISSUE THREAD

**Issue by marko-hologram** [twigphp/Twig#3470]
Greetings everyone!

**TL;DR**: Can be installed, but following the official example failed.

Twig fails to install using Composer and PHP 8 when using the suggested command from the [docs](https://twig.symfony.com/doc/3.x/installation.html) `composer require "twig/twig:^3.0"`.

It throws this error:
```
[InvalidArgumentException]
  Package twig/twig at version 3.0 has a PHP requirement incompatible with your PHP version, PHP extensions and Compo
  ser version
```
![image](https://user-images.githubusercontent.com/22904204/104353394-2132aa80-5508-11eb-8d85-e631419076a0.png)

This was ran using:
Windows 10
PHP v8.0.1
Composer v2.0.8

Also had it fail when I used an older Composer version, something like v1.10.x, but then it showed a different error where it mentioned the PHP >= 7.2.5 (or something like that, I don't have the exact error message anymore) dependency and that it was incompatible with my PHP 8.0.1 that I was using. At that point I noticed that I had the old Composer v1 so I decided to update.

---

But, there is some good news. Managed to install it by just running `composer require twig/twig` which I saw on the [Packagist page](https://packagist.org/packages/twig/twig) for Twig.

I also tried running `composer require "twig/twig:^3.2"`, but this fails with this error:
```
[InvalidArgumentException]
  Could not find package twig/twig in a version matching 3.2
```
![image](https://user-images.githubusercontent.com/22904204/104354281-4378f800-5509-11eb-9530-2cc87e5e800a.png)

I find this a bit odd because I would expect `^3.2` to match `3.2.x`, `3.3.x` and so on, but maybe I'm wrong, I don't really have any experience with Composer, only NPM.

**Comment by stof** [comment]
it says `matching 3.2` and not `matching ^3.2`. I suspect that the `^` gets lost due to some shell behavior (`^` is a special char in some windows terminal IIRC)

**Comment by marko-hologram** [comment]
> 
> 
> it says `matching 3.2` and not `matching ^3.2`. I suspect that the `^` gets lost due to some shell behavior (`^` is a special char in some windows terminal IIRC)

Ohhhh looks like you are right. I've been using PowerShell Core while trying all of this and now after your comment I decided to use regular old Command Prompt in Windows and ran `composer require "twig/twig:^3.0"` and it worked. Didn't think that `^` would get lost in quoted argument such as `"twig/twig:^3.0"`

Well, then this is kind of a false alarm, but definitely thanks for clearing it up 😄 

I guess I should just close the issue?
